### ISSUE TITLE
Commit category ignored

### ISSUE THREAD

**Issue by msolli** [p6spy/p6spy#469]
Hi! Thanks for a great library!

I've run into the same issue as #365.

Here's my spy.properties:
```
appender=com.p6spy.engine.spy.appender.Slf4JLogger
logMessageFormat=com.p6spy.engine.spy.appender.CustomLineFormat
customLogMessageFormat=[%(executionTime) ms] %(category):\n%(sql)
filter=true
exclude=UPDATE\\s+worker_jobs
reloadproperties=true
reloadpropertiesinterval=10
```

With this configuration, `commit` statements don't get logged. If I comment out `filter=true`, or set it to `filter=false`, they will get logged.

Let me know if you need more information about my setup.

**Comment by gavlyukovskiy** [comment]
If exclude filter is enabled all sqls go through the matcher to determine if it needs to be printed. In case of commit event `sql = ""` which doesn't pass effective regexp `(?mis)^(?!.*(UPDATE\s+worker_jobs).*)(.*)$` generated in `P6LogOptions#computeIncludeExcludePattern`. This happens because `^` in multiline mode will not match empty string (https://stackoverflow.com/questions/8896201/regular-expression-doesnt-match-empty-string-in-multiline-mode-java). Also same thing will happen if someone uses `include`.
@typekpb I think in terms of the logic we don't really have to have `^` and `$` in regexp builder.

**Comment by typekpb** [comment]
fixed in version 3.8.2, thanks to @gavlyukovskiy
